### PR TITLE
[node] Test all typesVersions

### PR DESCRIPTION
**Pull request by jablko**
Before #47195 the tests in the [test](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node/test) directory were only run on the oldest TypeScript versions (< 3.2), because they're listed in [types/node/tsconfig.json](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6e8dd401ba17be8b3941f48e03fc29a9495c5438%5E/types/node/tsconfig.json). After, they're only run on the newest TypeScript versions (> 3.6).

This change consistently runs the tests on all versions.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

File: types/node/ts3.4/tslint.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "rules": {
         "ban-types": false,
         "no-bad-reference": false,
+        "no-outside-dependencies": false,
```
Comment (jablko) [inline types/node/ts3.4/tslint.json:6]:
```suggestion

File: types/node/ts3.6/tslint.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
     "extends": "dtslint/dt.json",
     "rules": {
-        "no-bad-reference": false
+        "no-bad-reference": false,
+        "no-outside-dependencies": false
```
Comment (jablko) [inline types/node/ts3.6/tslint.json:5]:
```suggestion
        "no-bad-reference": false

File: types/node/v11/ts3.6/tslint.json [hunk 3]
```diff
@@ -3,6 +3,7 @@
     "rules": {
         "ban-types": false,
         "no-bad-reference": false,
+        "no-outside-dependencies": false,
```
Comment (jablko) [inline types/node/v11/ts3.6/tslint.json:6]:
```suggestion

File: types/node/v12/ts3.6/tslint.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "dtslint/dt.json",
   "rules": {
-    "no-bad-reference": false
+    "no-bad-reference": false,
+    "no-outside-dependencies": false
```
Comment (jablko) [inline types/node/v12/ts3.6/tslint.json:5]:
```suggestion
        "no-bad-reference": false

File: types/node/v13/ts3.4/tslint.json [hunk 5]
```diff
@@ -3,6 +3,7 @@
     "rules": {
         "ban-types": false,
         "no-bad-reference": false,
+        "no-outside-dependencies": false,
```
Comment (jablko) [inline types/node/v13/ts3.4/tslint.json:6]:
```suggestion

File: types/node/v13/ts3.6/tslint.json [hunk 6]
```diff
@@ -1,6 +1,7 @@
 {
     "extends": "dtslint/dt.json",
     "rules": {
-        "no-bad-reference": false
+        "no-bad-reference": false,
+        "no-outside-dependencies": false
```
Comment (jablko) [inline types/node/v13/ts3.6/tslint.json:5]:
```suggestion
        "no-bad-reference": false

**Review by jablko — COMMENTED** [review]
@jablko I noticed these differences from the required form. If you can revise your changes to avoid them, so much the better! Otherwise a maintainer will take a look.

If it's obvious why the differences are needed, feel free to "dismiss" this review. Thanks!

**Review by ExE-Boss — APPROVED** [review]
This needs rebasing.

File: .github/workflows/CI.yml [hunk 7]
```diff
@@ -13,7 +13,9 @@ jobs:
 
       # Get local dependencies
       - run: npm install
-
+      # https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49358/checks?check_run_id=1722333591#step:5:3341
+      - run: echo mongoose >> node_modules/@definitelytyped/dtslint-runner/expectedFailures.txt
```
Comment (sandersn) [inline .github/workflows/CI.yml:17]: can you please revert the two CI changes in this PR too?

**Review by sandersn — COMMENTED** [review]
Looks reasonable to me modulo the CI changes. Although I didn't review the individual test changes, I assume @ExE-Boss did.

(I closed this by mistake for about 45 seconds, sorry about that.)

File: .github/workflows/CI.yml (hunk 7 above)
Comment (jablko) [reply .github/workflows/CI.yml]: @sandersn Thanks, done :heavy_check_mark:

**Pull request merged**